domain/application/service/charm.go

```diff
@@ -137,6 +144,12 @@ type CharmState interface {
 	ListCharmsWithOriginByNames(ctx context.Context, names []string) ([]charm.CharmWithOrigin, error)
 }
 
+type CharmStore interface {
```

Comment:
May be worth putting a comment here, considering CharmStore is also a legacy term that means something else.

Reply:
yeah good point